kythe/rust/indexer/src/indexer/entries.rs

```diff
@@ -108,4 +108,37 @@ impl<'a> EntryEmitter<'a> {
         self.emit_node(anchor_vname, "/kythe/loc/end", byte_end.to_string().into_bytes().to_vec())?;
         self.emit_edge(anchor_vname, target_vname, "/kythe/edge/ref")
     }
+
+    /// Creates a diagnostic node with a tagged edge from an anchor or file
+    /// vname to the node Accepts a message, optional details, and an
+    /// optional url
+    ///
+    /// # Errors
+    /// If an error occurs while writing the entry, an error is returned.
+    pub fn emit_diagnostic<S: Into<String>>(
+        &mut self,
```

Comment:
What if you pulled the vname generation in here? Use the compilation unit hash plus some (in process) sequence to make sure the sigs are unique?